### PR TITLE
fix: set jamf plugin default checks to include PluginStaticCredentialsRef

### PR DESCRIPTION
**Pull request by flyinghermit**
Part of: https://github.com/gravitational/cloud/issues/4462

### CODE REVIEW

**Review by kimlisa — APPROVED** [review]

File: api/types/plugin.go [hunk 1]
```diff
@@ -153,8 +153,9 @@ func (p *PluginV1) CheckAndSetDefaults() error {
 		if p.Credentials == nil {
 			return trace.BadParameter("credentials must be set")
 		}
-		if p.Credentials.GetIdSecret().Id == "" || p.Credentials.GetIdSecret().Secret == "" {
-			return trace.BadParameter("Jamf plugin requires Jamf account username and password")
+		staticCreds := p.Credentials.GetStaticCredentialsRef()
+		if staticCreds == nil {
+			return trace.BadParameter("jamf plugin must be used with the static credentials ref type")
```
[the remark this answers is not in the document]
Comment (sshahcodes) [reply api/types/plugin.go:158]: Added in https://github.com/gravitational/teleport/pull/27616/commits/fc2e012f73f7fafbdc03baa2cac0736aa66e5955. Though that field is not in use currently (may be in future when we support credential rotation)

File: api/types/plugin.go [hunk 2]
```diff
@@ -147,14 +147,18 @@ func (p *PluginV1) CheckAndSetDefaults() error {
 			return trace.BadParameter("labels must be specified")
 		}
 	case *PluginSpecV1_Jamf:
-		if settings.Jamf.JamfSpec.ApiEndpoint == "" {
-			return trace.BadParameter("api endpoint must be set")
+		if err := settings.Jamf.CheckAndSetDefaults(); err != nil {
```
[the remark this answers is not in the document]
Comment (sshahcodes) [reply api/types/plugin.go:154]: I think yes, that would do catch all check. Added in https://github.com/gravitational/teleport/pull/27616/commits/cb806bbfffea066921027714b179aee575fc1834.

**Review by rosstimothy — APPROVED** [review]

File: api/types/plugin.go [hunk 3]
```diff
@@ -147,14 +147,22 @@ func (p *PluginV1) CheckAndSetDefaults() error {
 			return trace.BadParameter("labels must be specified")
 		}
 	case *PluginSpecV1_Jamf:
-		if settings.Jamf.JamfSpec.ApiEndpoint == "" {
-			return trace.BadParameter("api endpoint must be set")
+		// Check Jamf settings.
+		if settings.Jamf != nil {
+			return trace.BadParameter("missing Jamf settings")
+		}
```
Comment (ibeckermayer) [inline api/types/plugin.go:153]: I think this is meant to be

```go
if settings.Jamf == nil {
    return trace.BadParameter("missing Jamf settings")
}
```

Comment (sshahcodes) [reply api/types/plugin.go:153]: Oops! fix in https://github.com/gravitational/teleport/pull/27814. 
Many thanks!

**Pull request merged**